### PR TITLE
Jonathan - Fixed Weekly Summary Color Change logic

### PR DESCRIPTION
**Pull request by jonathancarpena**
# Description
- Update "Bio announcement", "Total Valid Weekly Summaries", and "Hours logged" to reflect Weekly Summary Option color

# Changes Made
- Previously, the weekly summary color option was being passed down as a parameter into the component however, the changes were not being reflected on both the Main and Dev site. 
- So, I updated the logic so that the colors would be generated within each function based on the **summary** parameter that gets passed down to each component.

## PR Related
- [#970](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/970), [#1026](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1026)

## How to test:
1. check into `fix_weeklySummaryColorChange`
2. do `npm install` and `npm run start:local` to run this PR locally

For Current User:
1. log as admin user
2. go to Profile page → Volunteer Times tab → Weekly Summary Options → Change Team
3. logged in as an Admin → Reports → Weekly Summary Reports
4. verify if the **summary text**, **Bio Announcement**, **Total Valid Weekly Summaries**, & **Hours Logged** has changed color to the corresponding Weekly Summary Option

For New User:
1. log as admin user
2. go to Other Links → User Management → Create New User → Weekly Summary Options → Choose a Team
3. go to the _user you created_ Profile page → Volunteer Times tab
4. verify if Weekly Summary Option is the one you selected
7. log in as new user and submit a weekly summary
8. logged in as an Admin → Reports → Weekly Summary Reports
9. verify if the **summary text**, **Bio Announcement**, **Total Valid Weekly Summaries**, & **Hours Logged** has changed color to the corresponding Weekly Summary Option

## Screenshot of Changes
PR # (2)
PR #1041 (3)

# Updated Logic
PR # (0)
PR # (1)

### CODE REVIEW

**Review by winghojackyli — APPROVED** [review]
Hi @jonathancarpena , I tested the PR and it worked as expected. 

### Current User
I logged in as admin and change my team to orange. Summary text, Bio Announcement, Total Valid Weekly Summaries, & Hours Logged has changed color to orange. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/9661f5f5-ff81-4388-98dc-dfd534201fd1

### New User
I created a new user with team blue. Then, I logged weekly summary report using the new user. Summary text, Bio Announcement, Total Valid Weekly Summaries, & Hours Logged are all in blue. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/6755deb8-edce-43ba-88e1-8ad3ab77f7ad

**Review by sxiong5 — APPROVED** [review]
Hey, I tested the PR, and it worked as intended.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62367016/198e9e23-1a84-4c22-b382-7087b8150614

**Comment by AriaYu927** [comment]
Hi @jonathancarpena , I've tested your PR and it works great!

Current user (Green): 
截屏2023-07-26 16 28 48

New user(blue):
截屏2023-07-26 16 32 33

截屏2023-07-26 16 33 55

**Review by eduvarjaor — APPROVED** [review]
Hi @jonathancarpena 

I tested this PR and everything working well. Good job!

1) Changed my team option
Screenshot_1

2) My team color reflecting in my weekly summaries report
Screenshot_2

3) Creating a new user with a predefined team
Screenshot_3

4) The team color reflecting in new user weekly summaries report
Screenshot_4

**Comment by iTvX** [comment]
Firstly, I use my admin account and then change team to Indigo, it works.
CleanShot 2023-07-26 at 12 45 34@2x
Then, I change my current use to team yellow, it works.
CleanShot 2023-07-26 at 12 47 23@2x
Then, I
CleanShot 2023-07-26 at 12 49 11@2x
 create a new user, set team green, it also works. 
Great job!

**Comment by one-community** [comment]
Thank you all, merging!

File: src/components/WeeklySummariesReport/FormattedReport.jsx [hunk 1]
```diff
@@ -121,18 +121,21 @@ const FormattedReport = ({ summaries, weekIndex, bioCanEdit }) => {
     );
   };
 
-  const getTotalValidWeeklySummaries = (summary, teamColor) => {
+  const getTotalValidWeeklySummaries = summary => {
+    const style = {
+      color: textColors[summary?.weeklySummaryOption] || textColors['Default'],
+    };
     if (summary.weeklySummariesCount === 8) {
```
Comment (iTvX) [inline src/components/WeeklySummariesReport/FormattedReport.jsx:128]: Why write two `getTotalValidWeeklySummaries` function?  They only differ based on the condition `summary.weeklySummariesCount === 8`, I think we can extract this part of the code:
`const getTotalValidWeeklySummaries = summary => {
  const style = {
    color: textColors[summary?.weeklySummaryOption] || textColors['Default'],
  };
  return (
    <p>
      <b style={style}>Total Valid Weekly Summaries:</b>{' '}
      {summary.weeklySummariesCount || 'No valid submissions yet!'}
    </p>
  );
};
`

**Pull request merged**